datafusion/proto/src/logical_plan.rs

```diff
@@ -625,11 +714,11 @@ impl AsLogicalPlan for LogicalPlanNode {
                 builder.build()
             }
             LogicalPlanType::Union(union) => {
-                let mut input_plans: Vec<LogicalPlan> = union
-                    .inputs
-                    .iter()
-                    .map(|i| i.try_into_logical_plan(ctx, extension_codec))
-                    .collect::<Result<_, DataFusionError>>()?;
+                let mut input_plans: Vec<LogicalPlan> = vec![];
```

Comment:
making something async and having to switch from a functional style is unfortunate -- one can also make streams but that syntax (well really the errors if you get it wrong) becomes wild